fix: set tracking branch only if not set

maxice8 opened the pull request:
**Description**
<!--- Describe your changes in detail -->
Split `git.SetUpstream` out of `git.Push`, the latter no longer runs with `--set-upstream` as to avoid overriding the tracking branch set by the user. Instead we set it with `git.SetUpstream` if none is set

**Related Issue**
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
Resolves #406 


**How Has This Been Tested?**
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Creating MRs on gitlab.alpinelinux.org, it doesn't override my tracking branches but more testing would be nice before merging.

**Types of changes**
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

maxice8 committed:
refactor(internal/git): Split Push into SetUpstream

maxice8 committed:
feat(commands/mr/create): set tracking remote if none is set

Don't push to tracking remote, there is a legitimate usecase that some
tracking remotes point to where it should not be pushed like
`upstream/master`.

maxice8 commented:
I made some changes so this is currently untested

profclems commented:
You'd have to fix the test. This is what is expected:

```sh
Branch 'mr-autofill-test-br' set up to track remote branch 'mr-autofill-test-br' from 'origin'.\n!1 docs: add some changes to txt file (mr-autofill-test-br)\n https://gitlab.com/glab-cli/test/-/merge_requests/1\n\n
```
But since the tracking branch is already set in the testdata, the output will be 
```sh
!1 docs: add some changes to txt file (mr-autofill-test-br)\n https://gitlab.com/glab-cli/test/-/merge_requests/1\n\n
```

maxice8 committed:
test(commands/mr/create): fix tests

git.Push no longer runs with --set-upstream so adjust for the new
expected output

maxice8 commented:
Seems like I can't access the codecov to find out the error

profclems commented:
It uses GitLab CI to run the tests. Check the `GitLab CI pipeline (pull request)` from the checks

profclems commented:
https://gitlab.com/profclems/glab/-/jobs/932748101

maxice8 commented:
> https://gitlab.com/profclems/glab/-/jobs/932748101

ty

maxice8 commented:
Pushed a new version that makes the `git branch --set-upstream` command be done after pushing, we need this because we can't do `git branch --set-upstream` to a branch that does not exist, so push (without using `--set-upstream`) then check if we are tracking and if not set the tracking branch.
